### ISSUE TITLE
Mastodon v4.0.0rc2 - The PUT request changing the image description returns a 404 

### ISSUE THREAD

**Issue by Sonstwer** [mastodon/mastodon#19921]
### Steps to reproduce the problem

1. I uploaded an image and sent it
2. I edited the toot to add an alt text. - That failed ("unexpected error")
3. I tried to delete and repost this toot containing an image and got some 404 errors.

### Expected behaviour

Image Alt Texts (Image Descriptions) should be editable

### Actual behaviour

they are not and throw errors including a 404 error

### Specifications

Mastodon v4.0.0rc2
Firefox 106.0.5 (64-bit)
Windows

**Comment by ClearlyClaire** [comment]
Editing images that are already attached to a published post is currently not possible.

**Comment by jcbeck37** [comment]
Having this issue on @mastodon.social - cannot edit Description of image or delete + re-draft and add Description to an image.

Firefox 106.0.3 (64-bit)
Windows 10

`PUT` https://mastodon.social/api/v1/media/109302702723167420 returns 404 Not Found.

**Comment by ClearlyClaire** [comment]
If I understand correctly, you're both trying to delete & redraft a post you tried to edit?

**Comment by ClearlyClaire** [comment]
What happens is that:
1. Editing media description is not possible yet
2. There is a bug that makes the web interface fail to exit edit mode when doing Delete & Redraft

https://github.com/mastodon/mastodon/pull/20023 should fix 2.
